tests/unit/completion/test_models.py

```diff
@@ -758,6 +758,34 @@ def test_tab_completion_not_sorted(qtmodeltester, fake_web_tab, win_registry,
     })
 
 
+def test_tab_completion_tabs_are_windows(qtmodeltester, fake_web_tab,
+                                         win_registry, tabbed_browser_stubs,
+                                         config_stub):
+    """Verify tabs across all windows are listed under a single category."""
+    tabbed_browser_stubs[0].widget.tabs = [
+        fake_web_tab(QUrl('https://github.com'), 'GitHub', 0),
+        fake_web_tab(QUrl('https://wikipedia.org'), 'Wikipedia', 1),
+        fake_web_tab(QUrl('https://duckduckgo.com'), 'DuckDuckGo', 2),
+    ]
+    tabbed_browser_stubs[1].widget.tabs = [
+        fake_web_tab(QUrl('https://wiki.archlinux.org'), 'ArchWiki', 0),
+    ]
+
+    config_stub.val.tabs.tabs_are_windows = True
+    model = miscmodels.buffer()
+    model.set_pattern('')
+    qtmodeltester.check(model)
+
+    _check_completions(model, {
+        'Windows': [
+            ('0/1', 'https://github.com', 'GitHub'),
+            ('0/2', 'https://wikipedia.org', 'Wikipedia'),
+            ('0/3', 'https://duckduckgo.com', 'DuckDuckGo'),
+            ('1/1', 'https://wiki.archlinux.org', 'ArchWiki'),
```

Comment:
Just to clarify, this is testing a case where the user has `tabs_are_windows=True`, yet there are still 3 tabs in the first window (because you can still have multiple tabs per window even with that setting)?

Reply:
Yes, that is indeed what the test is doing, and you're indeed correct that it's still possible to have 3 tabs in the first window even with `tabs_are_windows=True`. One way to achieve that would be to launch the browser with `tabs_are_windows=False`, open the three tabs, then execute `:set tabs.tabs_are_windows true`, then `:open -t <new-url>`, which will open <new-url> in a new window, while the 3 tabs from the first window remain as such.